File: app/src/main/java/org/deiverbum/app/model/BibleCommentList.java
package org.deiverbum.app.model;

import static org.deiverbum.app.utils.Constants.TITLE_BIBLE_COMMENTS;
import static org.deiverbum.app.utils.Utils.LS2;

import android.text.SpannableStringBuilder;

import com.google.firebase.firestore.PropertyName;

import org.deiverbum.app.utils.ColorUtils;
import org.deiverbum.app.utils.Utils;

import java.util.List;

public class BibleCommentList {
    public String padre;
    @SuppressWarnings("unused")
    public int id_homilia;
    public String pericopa;
    private Today today;
    public List<List<BibleComment>> allComentarios;
    private MassReading biblica;

    @PropertyName("comentarios")
    public List<BibleComment> comentarios;

    public BibleCommentList() {
    }


    @PropertyName("comentarios")
    public void setComentarios(List<BibleComment> comentarios) {
        this.comentarios = comentarios;
    }

    public void setAllComentarios(List<List<BibleComment>> comentarios) {
        this.allComentarios = comentarios;
    }

    @PropertyName("comentarios")
    public List<BibleComment> getComentarios() {
        return this.comentarios;
    }

    private SpannableStringBuilder getTitulo() {
        return Utils.toH3Red(TITLE_BIBLE_COMMENTS);
    }

    private String getTituloForRead() {
        return Utils.pointAtEnd(TITLE_BIBLE_COMMENTS);
    }


    public SpannableStringBuilder getAllForView(boolean nightMode) {
        ColorUtils.isNightMode=nightMode;
        SpannableStringBuilder sb = new SpannableStringBuilder();
        try {
            sb.append(today.getSingleForView());
            sb.append(LS2);
            sb.append(getTitulo());
            sb.append(LS2);
            for (List<BibleComment> subList : allComentarios) {
                if (subList.size() > 0) {
                    int x = 1;
                    for (BibleComment item : subList) {
                        if (x++ == 1) {
                            sb.append(item.getBiblica().getAll());
                            sb.append(LS2);
                            sb.append(Utils.formatTitle(TITLE_BIBLE_COMMENTS));
                            sb.append(LS2);
                        }
                        sb.append(item.getAllForView());
                    }
                }
            }
        } catch (Exception e) {
            sb.append(Utils.createErrorMessage(e.getMessage()));
        }
        return sb;
    }

    public StringBuilder getAllForRead() {
        StringBuilder sb = new StringBuilder();
        try {
            sb.append(today.getSingleForRead());
            sb.append(getTituloForRead());
            for (List<BibleComment> subList : allComentarios) {
                if (subList.size() > 0) {
                    int x = 1;
                    for (BibleComment item : subList) {
                        if (x++ == 1) {
                            sb.append(item.getBiblica().getAllForRead());
                            sb.append(Utils.pointAtEnd(TITLE_BIBLE_COMMENTS));
                        }
                        sb.append(item.getAllForRead());
                    }
                }
            }
        } catch (Exception e) {
            sb.append(Utils.createErrorMessage(e.getMessage()));

        }
        return sb;
    }


    @SuppressWarnings("unused")
    public void setPericopa(String pericopa) {
        this.pericopa = pericopa;
    }

    @SuppressWarnings("unused")
    public String getPericopa() {
        return this.pericopa;
    }

    public void setHoy(Today today) {
        this.today = today;
    }

    public MassReading getBiblica() {
        return biblica;
    }

    public void setBiblica(MassReading biblica) {
        this.biblica = biblica;
    }

}

File: app/src/main/java/org/deiverbum/app/model/BibleCommentList.kt
package org.deiverbum.app.model

import android.text.SpannableStringBuilder
import com.google.firebase.firestore.PropertyName
import org.deiverbum.app.domain.model.TodayRequest
import org.deiverbum.app.util.ColorUtils
import org.deiverbum.app.util.Constants
import org.deiverbum.app.util.Constants.ERR_NO_COMMENT
import org.deiverbum.app.util.Utils

class BibleCommentList {
    var padre: String? = null
    var allComentarios: MutableList<List<BibleComment?>> = ArrayList()

    @get:PropertyName("comentarios")
    @set:PropertyName("comentarios")
    @PropertyName("comentarios")
    var comentarios: List<BibleComment>? = null
    private var today: Today? = null
    var biblica: MassReading? = null
    var type = 0


    fun getAllForView(todayRequest: TodayRequest): SpannableStringBuilder {
        ColorUtils.isNightMode = todayRequest.isNightMode
        val sb = SpannableStringBuilder()
        try {

            //sb.append(Utils.LS2)

            for (subList in allComentarios) {
                if (subList.isNotEmpty()) {
                    var x = 1
                    for (item in subList) {
                        if (item!!.biblica!!.order!! > 39) {
                        if (x++ == 1) {
                            sb.append(item.biblica!!.getAll(type))
                            sb.append(Utils.LS2)
                            sb.append(
                                Utils.toH1Red(
                                    Constants.TITLE_BIBLE_COMMENTS))
                            sb.append(Utils.LS2)
                        }
                        sb.append(item.allForView)

                    }
                    }

                }
            }
        } catch (e: Exception) {
            sb.append(Utils.createErrorMessage(e.message))
        }
        if(sb.isEmpty()){
            sb.append(ERR_NO_COMMENT)
        }
        return sb
    }

    val getAllForRead: StringBuilder
        get() {
            val sb = StringBuilder()
            try {
                //sb.append(today!!.getSingleForRead())
                //sb.append(tituloForRead)
                for (subList in allComentarios) {
                    if (subList.isNotEmpty()) {
                        var x = 1
                        for (item in subList) {
                            if (x++ == 1) {
                                sb.append(item?.biblica!!.getAllForRead(type))
                                sb.append(
                                    Utils.pointAtEnd(
                                        Constants.TITLE_BIBLE_COMMENTS))
                            }
                            sb.append(item?.allForRead)
                        }
                    }
                }
            } catch (e: Exception) {
                sb.append(Utils.createErrorMessage(e.message))
            }
            return sb
        }

    fun setHoy(today: Today?) {
        this.today = today
    }
}